theseus/core/cost_function.py

```diff
@@ -166,61 +166,56 @@ def _compute_error(
     def error(self) -> torch.Tensor:
         return self._compute_error()[0]
 
+    def _make_jac_fn(
+        self, tmp_optim_vars: Tuple[Manifold, ...], tmp_aux_vars: Tuple[Variable, ...]
+    ) -> Callable:
+        def jac_fn(*optim_vars_data_):
+            assert len(optim_vars_data_) == len(tmp_optim_vars)
+            for i, tensor in enumerate(optim_vars_data_):
+                tmp_optim_vars[i].update(tensor)
+
+            return self._err_fn(optim_vars=tmp_optim_vars, aux_vars=tmp_aux_vars)
+
+        return jac_fn
+
+    def _compute_autograd_jacobian(
+        self, optim_tensors: Tuple[torch.Tensor, ...], jac_fn: Callable
+    ) -> Tuple[torch.Tensor, ...]:
+        return autogradF.jacobian(
+            jac_fn,
+            optim_tensors,
+            create_graph=True,
+            strict=self._autograd_strict,
+            vectorize=self._autograd_vectorize,
+        )
+
     # Returns (jacobians, error)
     def jacobians(self) -> Tuple[List[torch.Tensor], torch.Tensor]:
         err, optim_vars, aux_vars = self._compute_error()
 
         if not self._autograd_loop_over_batch:
```

Comment:
This is a bit of a nit, but it's usually good practice to avoid negative conditionals like `if not something:` because the positive version is easier to parse/understand by readers.